cmd/register/cmd/discovery.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2015 Sorint.lab
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+)
+
+// ServiceDiscovery helps to register service
+type ServiceDiscovery interface {
+	Register(info *ServiceInfo) error
+}
+
+// NewServiceDiscovery creates a Discovery from registerBackend and registerEndpoints
+func NewServiceDiscovery(registerBackend string, registerEndpoints string) (ServiceDiscovery, error) {
+	switch registerBackend {
+	case "consul":
+		return NewConsulServiceDiscovery(&http.Client{}, registerEndpoints), nil
+	default:
+		return nil, errors.New("register backend not supported")
+	}
+}
+
+// A HTTPClient has necessary method to make http calls
```

Comment:
Can't you just use consul client lib (I'm not asking for libkv since I'd like to ditch it)? In this way it'll be easier to handle context, tls etc... It's already vendored (perhaps needs an update to the latest version).

Reply:
sure, will make necessary changes to use consul client lib instead of HTTP calls